service/src/main/java/uk/gov/hmcts/reform/fpl/controllers/SendDocumentController.java

```diff
@@ -0,0 +1,79 @@
+package uk.gov.hmcts.reform.fpl.controllers;
+
+import com.fasterxml.jackson.core.type.TypeReference;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import io.swagger.annotations.Api;
+import lombok.RequiredArgsConstructor;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RestController;
+import uk.gov.hmcts.reform.ccd.client.model.AboutToStartOrSubmitCallbackResponse;
+import uk.gov.hmcts.reform.ccd.client.model.CallbackRequest;
+import uk.gov.hmcts.reform.ccd.client.model.CaseDetails;
+import uk.gov.hmcts.reform.fpl.model.CaseData;
+import uk.gov.hmcts.reform.fpl.model.DocumentsSentToParty;
+import uk.gov.hmcts.reform.fpl.model.Representative;
+import uk.gov.hmcts.reform.fpl.model.SentDocument;
+import uk.gov.hmcts.reform.fpl.model.common.DocumentReference;
+import uk.gov.hmcts.reform.fpl.model.common.Element;
+import uk.gov.hmcts.reform.fpl.service.DocumentSenderService;
+import uk.gov.hmcts.reform.fpl.service.RepresentativeService;
+import uk.gov.hmcts.reform.fpl.service.SentDocumentHistoryService;
+
+import java.util.List;
+
+import static uk.gov.hmcts.reform.fpl.enums.RepresentativeServingPreferences.POST;
+
+@Api
+@RestController
+@RequestMapping("/callback/send-document")
+@RequiredArgsConstructor(onConstructor = @__(@Autowired))
+public class SendDocumentController {
```

Comment:
when is the best time to write integration test for this? as part of this PR or when we have everything together? need to check that documentsSentToParties is updated in CaseData

Reply:
there is test for it, you even commented on it, see SendDocumentEventControllerTest

Reply:
oh I am blind 🤦‍♂